Added Director Nav Links

Jacob-GHub opened the pull request:
Added new routes and ignored the initial "director" section. I followed the routes on figma, but I was unsure if this was the correct route for the "directors" section

<img width="1511" alt="Screenshot 2024-11-05 at 2 38 15 PM" src="https://github.com/user-attachments/assets/63365cf0-7d45-4c75-aa2b-2c5e6695843a">

The Routes: src > data > director
``` tsx
export const links = [
  {
    name: "Director",
    link: "",
  },
  {
    name: "Applications",
    link: "/director/:program/application",
  },
  {
    name: "Leads",
    link: "/director/leads",
  },
  {
    name: "Directors",
    link: "/director/:program/profile/:uid",
  },
  {
    name: "Assignments",
    link: "/director/:program/assignments",
  },
  {
    name: "View Teams",
    link: "/director/:program/view",
  },
  {
    name: "Email",
    link: "/director/contacts",
  },
];
```

Jacob-GHub committed:
added flex

Jacob-GHub committed:
added routes

Jacob-GHub committed:
formatted

shahdivyank commented:
attach issue

shahdivyank commented on src/app/director/[program]/profile/[uid]/page.tsx:
```diff
@@ -0,0 +1,9 @@
+const Page = () => {
```
do not add the pages yet, just need the navigation

shahdivyank commented on src/data/director/navigation.tsx:
```diff
@@ -1,6 +1,30 @@
+  },
+  {
+    name: "Applications",
+    link: "/director/:program/application",
```
if the link is dynamic, we should instead make each link a function which returns a string for the dynamically generated url

Jacob-GHub committed:
removed additional pages and made links a function

Jacob-GHub committed:
added links and got rid of additional pages

Jacob-GHub committed:
used pathname params

Jacob-GHub committed:
retrigger checks

Jacob-GHub committed:
retrigger checks 2

shahdivyank commented on src/app/director/layout.tsx:
```diff
@@ -6,9 +9,16 @@ type LayoutProps = {
+  const segments = pathname.split("/");
+  const navigation = links.map(({ name, link }) => ({
+    name,
+    link: link(segments[2], segments[4]),
```
what happens when segments 2 and 4 dont exist? there should be a fallback

shahdivyank commented on src/data/director/navigation.tsx:
```diff
@@ -1,6 +1,31 @@
+  },
+  {
+    name: "Applications",
+    link: (program: string) => `/director/${program}/application`,
```
prolly want all the functions to be the same parameter wise so it gives less problems

Jacob-GHub committed:
Made parameters consistent and added a fallback of I
gnite and uid123, also made it so that static link segments wont feed into dynamic links

Jacob-GHub committed:
allowed unused arguments starting with _'s

shahdivyank commented on src/app/director/layout.tsx:
```diff
@@ -1,14 +1,29 @@
-    <div className="h-full">
-      <Navigation links={links} />
+    <div className="flex h-full">
+      <Navigation links={navigation} />
```
logic should be handled inside navigation

Jacob-GHub committed:
moved logic to navigation

Jacob-GHub committed:
Merge branch 'dev' into Jacob-GHub/Director-Nav-Links

seanquiambao committed:
Merge branch 'dev' of https://github.com/acm-ucr/starlight into Jacob-GHub/Director-Nav-Links

seanquiambao committed:
Merge branch 'dev' of https://github.com/acm-ucr/starlight into Jacob-GHub/Director-Nav-Links

seanquiambao committed:
added react hooks back?

seanquiambao committed:
eslint warning

seanquiambao committed:
eslint disable

seanquiambao committed:
i think thats it?
